spark/src/main/scala/com/cloudera/livy/spark/client/ClientSession.scala

```diff
@@ -19,31 +19,46 @@ package com.cloudera.livy.spark.client
 import java.net.URI
 import java.nio.ByteBuffer
 import java.util.concurrent
-import java.util.concurrent.{ExecutionException, TimeUnit}
+import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicLong
 
+import org.apache.spark.SparkConf
+
 import scala.concurrent.{ExecutionContext, Future}
 import scala.collection.JavaConverters._
 import scala.collection.mutable
 import scala.util.control.NonFatal
 
-import com.cloudera.livy.{Job, LivyClient}
+import com.cloudera.livy.{Logging, LivyClientBuilder}
 import com.cloudera.livy.client.local.LocalClient
 import com.cloudera.livy.sessions.{Session, SessionState}
 
-class ClientSession(val sessionId: Int, createRequest: CreateClientRequest) extends Session {
+class ClientSession(val sessionId: Int, createRequest: CreateClientRequest)
+  extends Session with Logging {
   implicit val executionContext = ExecutionContext.global
+
   var sessionState: SessionState = SessionState.Starting()
-  SessionClientTracker.createClient(
-    sessionId, createRequest.sparkConf.asJava, createRequest.timeout)
+
+  private val _timeout = createRequest.timeout
```

Comment:
You can declare `timeout` in the parent class as a `val` and avoid the indirection here.